activejob/lib/active_job/queue_adapters/delayed_job_adapter.rb

```diff
@@ -35,12 +35,22 @@ def initialize(job_data)
         end
 
         def display_name
-          "#{job_data['job_class']} [#{job_data['job_id']}] from DelayedJob(#{job_data['queue_name']}) with arguments: #{job_data['arguments']}"
+          base_name = "#{job_data['job_class']} [#{job_data['job_id']}] from DelayedJob(#{job_data['queue_name']})"
+
+          return base_name unless log_arguments?
+
+          "#{base_name} with arguments: #{job_data['arguments']}"
         end
 
         def perform
           Base.execute(job_data)
         end
+
+        private
+
+        def log_arguments?
+          Object.const_get(job_data['job_class']).log_arguments?
```

Comment:
We should probably match how `job_class` is constantised elsewhere:

https://github.com/rails/rails/blob/c3664b0c2b509fc0a47929f42afbd716959c8f4c/activejob/lib/active_job/core.rb#L61

```suggestion
          job_data["job_class"].constantize.log_arguments?
```

(even though they're [ultimately equivalent](https://github.com/rails/rails/blob/c3664b0c2b509fc0a47929f42afbd716959c8f4c/activesupport/lib/active_support/inflector/methods.rb#L280).)

This will necessitate adding a require to this file:

```ruby
require "active_support/core_ext/string/inflections"
```

Reply:
Cool, thanks @eugeneius, that's done now. Is it preferable to rebase and tidy up commits or leave as is?